src/crossroads/main.py

```diff
@@ -88,24 +88,21 @@ def __init__(self, location):
         self.longitude = location["lng"]
         self.infobox = location["infobox"]
 
-def backend():
-    df_gps = pd.read_json('sample.json')
-    print('There are {:,} rows in the location history dataset'.format(len(df_gps)))
+def json_parser():
+    """
+    Parse .json file with Location History from Google Maps timeline.
 
-    # parse lat, lon, and timestamp from the dict inside the locations column
-    df_gps['lat'] = df_gps['locations'].map(lambda x: x['latitudeE7'])
-    df_gps['lon'] = df_gps['locations'].map(lambda x: x['longitudeE7'])
-    df_gps['timestamp_ms'] = df_gps['locations'].map(lambda x: x['timestampMs'])
+    :param jfile: JSON file
+    :return: a list of chronologically arranged dictionaries, from the oldest ones
+    """
+    #test version with "sample.json" - swap TODO
+    x = open("sample.json", "r")
+    locations = json.loads(x.read())['locations']
 
-    # convert lat/lon to decimalized degrees and the timestamp to date-time
-    df_gps['lat'] = df_gps['lat'] / 10.**7
-    df_gps['lon'] = df_gps['lon'] / 10.**7
-    df_gps['timestamp_ms'] = df_gps['timestamp_ms'].astype(float) / 1000
-    df_gps['datetime'] = df_gps['timestamp_ms'].map(lambda x: dt.fromtimestamp(x).strftime('%Y-%m-%d %H:%M:%S'))
-    date_range = '{}-{}'.format(df_gps['datetime'].min()[:4], df_gps['datetime'].max()[:4])
+    for location in locations:
+        location['latitudeE7'] = location['latitudeE7'] / 10.**7
+        location['longitudeE7'] = location['longitudeE7'] / 10.**7
+        location['timestampMs'] = datetime.fromtimestamp(int(location['timestampMs'])/1000)
```

Comment:
This fields extraction with normalisation is fine, but you never assign these to new dictionary. You need to add something like this:
```        coordinates_with_timestamp = location
        parsed_location.append(coordinates_with_timestamp```
Keep in mind this is just an example and I am not sure if it will work out of the box.